Issue #270 spack disable local config

andrewmogan opened the pull request:
This PR seeks to address Issue #270 by 
1. setting `SPACK_DISABLE_LOCAL_CONFIG=true` when `env.sh` is sourced, 
2. doing the same in `dbt_create.py` when `-s` is given as an argument, 
3. setting `SPACK_USER_CACHE_PATH` to the dev environment so that `$HOME/.spack/cache` isn't created, and 
4. printing warning is printed if `$HOME/.spack` already exists

andrewmogan committed:
Print warning if user spack directory exists

andrewmogan committed:
Set SPACK_DISABLE_LOCAL_CONFIG=true

andrewmogan committed:
Add explanatory comment

andrewmogan committed:
Set spack environment variables to avoid conflicts with local environment

andrewmogan committed:
Default the user spack cache path to /tmp

andrewmogan committed:
No longer export spack user cache path here

andrewmogan committed:
Set spack environment variables in a more sensible way

ron003 commented:
I was expecting to see `export SPACK_USER_CACHE_PATH=...` in
env.sh
As for the value, did you verify that the variable value is the path to the parent dir into which the `cache` (sub)dir goes? This, as opposed to "the path to the cache" i.e. the name of the actual cache dir is contained in the variable value such that it could be named "apple" for example.    I'm guessing the former will be the answer.

andrewmogan committed:
Avoid SPACK_USER_CACHE_PATH being in $HOME

SPACK_USER_CACHE_PATH will default to /tmp/$USER/spack (will be created
if it doesn't exist). Once the working area env.sh is sourced, it will
be set to $DBT_AREA_ROOT

andrewmogan committed:
Merge branch 'develop' of https://github.com/DUNE-DAQ/daq-buildtools into amogan/issue270_spack_disable_local

andrewmogan committed:
Fix merge conflicts

andrewmogan commented:
The reason I didn't initially `export SPACK_USER_CACHE_PATH=${DBT_AREA_ROOT}` in `env.sh` is because `${DBT_AREA_ROOT}` isn't defined until you source `my/daq/area/env.sh`, which exports `DBT_AREA_ROOT=/my/daq/area`. So the new solution is to initially default `SPACK_USER_CACHE_PATH` to `/tmp/$USER/spack` (which does create `cache` under the `spack` directory, as per your question), and then export `SPACK_USER_CACHE_DIR=${DBT_AREA_ROOT}` when you run `my/daq/area/env.sh`. This means that `$HOME/.spack` is never created by `dbt-create -s`, but it will still print a warning if that directory exists. 

Here's a test that helps illustrate what I mean. I ran this on `daq.fnal.gov` as myself. After cloning `daq-buildtools` and checking out this branch,
```
$ pwd # /nfs/home/amogan
$ env | grep SPACK # Initially empty

$ ls $HOME/.spack # If exists, (re)move it

$ source daq-buildtools/env.sh
$ env | grep SPACK 
SPACK_DISABLE_LOCAL_CONFIG=true
SPACK_USER_CACHE_PATH=/tmp/amogan/spack
$ ls /tmp/$USER/spack # Should be empty or nonexistent

$ dbt-create -s -n NFD_DEV_240411_A9 test_spack_disable_local
$ ls /tmp/$USER/spack # Now has cache subdir
$ ls $HOME/.spack # Should still not exist

$ cd test_spack_disable_local 
$ source env.sh
$ env | grep SPACK
SPACK_DISABLE_LOCAL_CONFIG=true
SPACK_USER_CACHE_PATH=/home/nfs/amogan/test_spack_disable_local
$ ls $HOME/.spack # Still nonexistent 
```

ron003 commented:
I'm confused by:  source daq-buildtools/env.sh

--Ron

On Fri, 2024-04-12 at 11:20 -0700, Andrew Mogan wrote:

[EXTERNAL] – This message is from an external sender

The reason I didn't initially export SPACK_USER_CACHE_PATH=${DBT_AREA_ROOT} inenv.sh is because ${DBT_AREA_ROOT} isn't defined until you sourcemy/daq/area/env.sh, which exports DBT_AREA_ROOT=/my/daq/area. So the new solution is to initially default SPACK_USER_CACHE_PATH to /tmp/$USER/spack (which does createcache under the spack directory, as per your question), and then exportSPACK_USER_CACHE_DIR=${DBT_AREA_ROOT} when you runmy/daq/area/env.sh. This means that $HOME/.spack is never created by dbt-create -s, but it will still print a warning if that directory exists.

Here's a test that helps illustrate what I mean. I ran this on daq.fnal.gov as myself. After cloning daq-buildtools and checking out this branch,

$ pwd # /nfs/home/amogan
$ env | grep SPACK # Initially empty

$ ls $HOME/.spack # If exists, (re)move it

$ source daq-buildtools/env.sh
$ env | grep SPACK
SPACK_DISABLE_LOCAL_CONFIG=true
SPACK_USER_CACHE_PATH=/tmp/amogan/spack
$ ls /tmp/$USER/spack # Should be empty or nonexistent

$ dbt-create -s -n NFD_DEV_240411_A9 test_spack_disable_local
$ ls /tmp/$USER/spack # Now has cache subdir
$ ls $HOME/.spack # Should still not exist

$ cd test_spack_disable_local
$ source env.sh
$ env | grep SPACK
SPACK_DISABLE_LOCAL_CONFIG=true
SPACK_USER_CACHE_PATH=/home/nfs/amogan/test_spack_disable_local
$ ls $HOME/.spack # Still nonexistent


—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_DUNE-2DDAQ_daq-2Dbuildtools_pull_272-23issuecomment-2D2052250322&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=7KXkqf0bzyUQv15jBKEcHw&m=olHZxZo7TAFT78vQHcjNcfsp7sm4fR_crMpuZw_jq1vFn0Qijjib-wb91VeXEBYQ&s=EGvD5sbfpIov49RhPIfPy853fGwIye-XZHX-tNxFxh4&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AGH7VZW7J7WJCCDBZCXLGTDY5AQXRAVCNFSM6AAAAABGC2ZRHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANJSGI2TAMZSGI&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=7KXkqf0bzyUQv15jBKEcHw&m=olHZxZo7TAFT78vQHcjNcfsp7sm4fR_crMpuZw_jq1vFn0Qijjib-wb91VeXEBYQ&s=BzRLDDbXBpwQhL34eMgPqYv39RUH5Cn5rftE01h1HDs&e=>.
You are receiving this because your review was requested.Message ID: ***@***.***>


--

Ron Rechenmacher
Engineer, Tools and Advanced Computing
Fermi National Accelerator Laboratory
Batavia, IL  60510

andrewmogan committed:
Move export of SPACK_USER_CACHE_PATH

Moved from ${DBT_ROOT}/env.sh to
${DBT_ROOT}/scripts/dbt-install-spack.sh. This prevents unncessary
environment variable exports in the case where dbt-create is called
without -s
